crates/bevy_math/src/common_traits.rs

```diff
@@ -161,3 +161,133 @@ impl NormedVectorSpace for f32 {
         self * self
     }
 }
+
+/// A type that can be intermediately interpolated between two given values
+/// using an auxiliary linear parameter.
```

Comment:
The parameter itself is "linear" in the sense that it is one-dimensional, and the second trait requirement treats the parameter itself linearly, since it involves interpolating `f32`s with `lerp` via `interpolate`, but this doesn't mean at all that the interpolation has to be linear. 

That assumption is more like a constant-velocity requirement; I'm not sure if I actually like it, but its *raison d'être* is that it guarantees that `smooth_nudge`'s decay parameter actually means something relatively precise.

Reply:
> The parameter itself is "linear" in the sense that it is one-dimensional,

That's "scalar" not "linear".

> and the second trait requirement treats the parameter itself linearly, since it involves interpolating f32s with lerp via interpolate, but this doesn't mean at all that the interpolation has to be linear.

I don't understand this sentence, you sem to contradict yourself. If the implementation is `lerp` then the only source of non linearity is `t` so it's not linear.